plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -469,6 +462,20 @@ private function process_select_field( $field_data, $options ) {
 		}
 
 		$field_data['options'] = $cleaned_options;
+
+		// If the field is not required, inject an empty option at the start.
+		if ( isset( $options['required'] ) && false === $options['required'] && ! in_array( '', $added_values, true ) ) {
```

Comment:
We need to get `$field_data['required']` instead here, $options doesn't contain required